src/Makefile.common

```diff
@@ -274,7 +278,13 @@ new: $(MAKEFILE_LIST)
 	-rm -f  $(MODULE_OBJECTS)
 	-rm -f  $(MODULE_FILES)
 	-rm -f  $(EXE)
-	$(MAKE) $(EXE) MAKELEVEL=0 -f $(MAKEFILE_LIST)
+	$(MAKE) $(MODULE_FILES)  # also makes MODULE_OBJECTS
```

Comment:
Is it maybe better to split up the module compilation, the regular source compilation, and the linking into separate targets and use those here?  I am just worried that there is some duplicate rules here.